app/cdap/components/PipelineDiff/index.tsx

```diff
@@ -43,30 +46,55 @@ const DiffContentContainer = ({
   version,
   latestVersion,
 }: IDiffContentContainerProps) => {
-  const dispatch = useDispatch();
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
-    fetchPipelineConfig(namespace, appId, version, latestVersion, dispatch);
+    dispatch(actions.fetchPipelinesPending());
+    fetchPipelineConfig({
+      namespace,
+      appId,
+      topVersion: version,
+      bottomVersion: latestVersion,
+    }).subscribe(
+      ({ topPipelineConfig, bottomPipelineConfig }) => {
+        const stages = topPipelineConfig.stages.concat(bottomPipelineConfig.stages);
+
+        fetchExtraPluginProperties({ namespace, stages }).subscribe(
```

Comment:
may consider refactoring nested subscribe with `.pipe()` and `.switchMap()`

Reply:
done.